lib/puppet/functions/validate_domain_name.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# @summary
+#   Validate that all values passed are syntactically correct domain names.
+#   Fail compilation if any value fails this check.
+Puppet::Functions.create_function(:validate_domain_name) do
```

Comment:
I'm still keen on all functions ported to the new API are available as namespaced versions.
The approach I've been taking is to add a non-namespaced deprecated shim.  See https://github.com/puppetlabs/puppetlabs-stdlib/pull/1341

(We also took this approach originally in extlib.  The deprecated versions were eventually dropped in a major release)

Reply:
+1

Reply:
I would like to do this after the whole refactoring where I fix non-legacy functions that relied on legacy functions, part of #1346.  So resolving for now.